bundles/jcr/contentparser/src/main/java/org/apache/sling/jcr/contentparser/Content.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.jcr.contentparser;
+
+import java.util.Map;
+
+/**
+ * Represents a resource or node in the content hierarchy.
+ */
+public interface Content {
```

Comment:
Can't we just rely on org.apache.sling.api.resource.Resource and throw a https://docs.oracle.com/javase/8/docs/api/java/lang/UnsupportedOperationException.html for the methods which are not useful in this  context, which are
- `getResourceMetadata()`
- `getResourceResolver()`
maybe `isResourceType(String resourceType)`.